android/sdl_android/src/main/java/com/smartdevicelink/managers/video/VideoStreamManager.java

```diff
@@ -645,6 +774,101 @@ void createTouchScalar(ImageResolution resolution, DisplayMetrics displayMetrics
+                        || capability.getPreferredResolution().getResolutionWidth() == null) {
+                    continue;
+                }
+                if (capability.getDiagonalScreenSize() == null) {
+                    diagonal = parameters.getPreferredDiagonal();
+                } else {
+                    diagonal = capability.getDiagonalScreenSize();
+                }
+
+                if (range.getMinScreenDiagonal() > diagonal) {
+                    continue;
+                }
+
+                if (!isAspectRatioInRange(range.getAspectRatio(), capability.getPreferredResolution())) {
+                    if (constraintHeightMax == null && constraintHeightMin == null) {
+                        continue;
+                    }
+                }
+
+                if (!isImageResolutionInRange(range.getMinResolution(), range.getMaxResolution(), capability.getPreferredResolution())) {
+                    continue;
+                }
+
+                validCapabilities.add(capability);
+            }
+        }
+
+        return validCapabilities;
+    }
+
+    public Boolean isImageResolutionInRange(Resolution minResolution, Resolution maxResolution, ImageResolution currentResolution) {
+
+        Integer constraintHeightMax = maxResolution.getResolutionHeight();
+        Integer constraintHeightMin = minResolution.getResolutionHeight();
+        Integer constraintWidthMax = maxResolution.getResolutionWidth();
+        Integer constraintWidthMin = minResolution.getResolutionWidth();
+        Integer resolutionHeight = currentResolution.getResolutionHeight();
+        Integer resolutionWidth = currentResolution.getResolutionWidth();
+        if (currentResolution.getResolutionHeight() > 0 && currentResolution.getResolutionWidth() > 0 && constraintHeightMax != null && constraintHeightMin != null) {
+            if (!(resolutionHeight >= constraintHeightMin && resolutionHeight <= constraintHeightMax)) {
+                return false;
+            }
+
+            if (!(resolutionWidth >= constraintWidthMin && resolutionWidth <= constraintWidthMax)) {
+                return false;
+            }
+        }
+
+        // TODO check what if dev provided invalid constraints
+        return true;
+    }
+
+    public Boolean isAspectRatioInRange(AspectRatio aspectRatio, ImageResolution currentResolution) {
+        Double aspectRatioMin = aspectRatio.getMinAspectRatio();
+        Double aspectRatioMax = aspectRatio.getMaxAspectRatio();
+
+        Double currentAspectRatio = Double.valueOf(currentResolution.getResolutionWidth()) / Double.valueOf(currentResolution.getResolutionHeight());
+
+        if (!(aspectRatioMax > aspectRatioMin && aspectRatioMin > 0)) {
+            if (!(currentAspectRatio >= aspectRatioMin && currentAspectRatio <= aspectRatioMax)) {
+                return false;
+            }
+        }
+        return true;
+    }
```

Comment:
For this logic we should be verifying that the ranges constraints are correct and then if the current aspect ratio meets those constraints. with the current implementation we are only checking the currentAspectRatio if the constraints are incorrect

```
        if (aspectRatioMax > aspectRatioMin && aspectRatioMin > 0) {
            return currentAspectRatio >= aspectRatioMin && currentAspectRatio <= aspectRatioMax;
        }
        else {
            return false;
        }
```